Sources/ConfidenceProvider/ConfidenceFeatureProvider.swift

```diff
@@ -260,43 +260,41 @@ public class ConfidenceFeatureProvider: FeatureProvider {
             throw OpenFeatureError.invalidContextError
         }
 
-        do {
-            let resolverResult = try resolver.resolve(flag: path.flag, ctx: ctx)
+        let resolverResult = try resolver.resolve(flag: path.flag, ctx: ctx)
 
-            guard let value = resolverResult.resolvedValue.value else {
-                return resolveFlagNoValue(
-                    defaultValue: defaultValue,
-                    resolverResult: resolverResult,
-                    ctx: ctx
-                )
-            }
+        guard let value = resolverResult.resolvedValue.value else {
+            return resolveFlagNoValue(
+                defaultValue: defaultValue,
+                resolverResult: resolverResult,
+                ctx: ctx
+            )
+        }
 
-            let pathValue: Value = try getValue(path: path.path, value: value)
-            guard let typedValue: T = pathValue == .null ? defaultValue : pathValue.getTyped() else {
-                throw OpenFeatureError.parseError(message: "Unable to parse flag value: \(pathValue)")
-            }
+        let pathValue: Value = try getValue(path: path.path, value: value)
+        guard let typedValue: T = pathValue == .null ? defaultValue : pathValue.getTyped() else {
+            throw OpenFeatureError.parseError(message: "Unable to parse flag value: \(pathValue)")
+        }
 
-            let evaluationResult = ProviderEvaluation(
+        if resolverResult.stale {
+            return ProviderEvaluation(
                 value: typedValue,
                 variant: resolverResult.resolvedValue.variant,
-                reason: Reason.targetingMatch.rawValue
-            )
-
-            processResultForApply(
-                resolverResult: resolverResult,
-                ctx: ctx,
-                applyTime: Date.backport.now
+                reason: Reason.stale.rawValue
             )
-            return evaluationResult
-        } catch ConfidenceError.cachedValueExpired {
-            return ProviderEvaluation(
-                value: defaultValue,
-                variant: nil,
-                reason: Reason.error.rawValue,
-                errorCode: ErrorCode.providerNotReady)
-        } catch {
-            throw error
         }
+
+        let evaluationResult = ProviderEvaluation(
+            value: typedValue,
+            variant: resolverResult.resolvedValue.variant,
+            reason: Reason.targetingMatch.rawValue
+        )
+
+        processResultForApply(
+            resolverResult: resolverResult,
+            ctx: ctx,
+            applyTime: Date.backport.now
+        )
+        return evaluationResult
```

Comment:
I realized something is wrong here, we need to call `processResultForApply` in case of STALE now

Reply:
Good point, I didn't think of it!